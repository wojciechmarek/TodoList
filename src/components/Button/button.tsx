import { ButtonStyle } from './button.styled';

export const Button = () => {
  return <ButtonStyle>Button</ButtonStyle>;
};
