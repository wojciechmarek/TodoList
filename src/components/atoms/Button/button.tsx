import { ButtonProps } from './button.interface';
import { ButtonStyle } from './button.styled';

export const Button = (props: ButtonProps) => {
  const { children, ...params } = props;
  return <ButtonStyle {...params}>{children}</ButtonStyle>;
};
