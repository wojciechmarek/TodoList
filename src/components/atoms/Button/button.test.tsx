import { configure, render } from '@testing-library/react';
import { ButtonProps } from './button.interface';
import { Button } from './button';
import '@testing-library/jest-dom';

describe('Button', () => {
  let buttonProps: ButtonProps;

  beforeEach(() => {
    configure({ throwSuggestions: true });

    buttonProps = {
      children: 'Button',
      onClick: jest.fn(),
    };
  });

  it('should render', () => {
    const { baseElement } = render(<Button {...buttonProps} />);
    expect(baseElement).toBeTruthy();
  });

  describe('when the button is rendered', () => {
    it('should have the correct text', () => {
      const { getByRole } = render(<Button {...buttonProps} />);
      const button = getByRole('button', { name: /button/i });
      expect(button).toHaveTextContent('Button');
    });
  });

  describe('when the button is clicked', () => {
    it('should call the onClick function', () => {
      const { getByRole } = render(<Button {...buttonProps} />);
      const button = getByRole('button', { name: /button/i });
      button.click();
      expect(buttonProps.onClick).toHaveBeenCalled();
    });
  });

  describe('when the button is disabled', () => {
    it('should not call the onClick function', () => {
      const { getByRole } = render(<Button {...buttonProps} disabled />);
      const button = getByRole('button', { name: /button/i });
      button.click();
      expect(buttonProps.onClick).not.toHaveBeenCalled();
    });

    it('should have the disabled attribute', () => {
      const { getByRole } = render(<Button {...buttonProps} disabled />);
      const button = getByRole('button', { name: /button/i });
      expect(button).toBeDisabled();
    });
  });
});
