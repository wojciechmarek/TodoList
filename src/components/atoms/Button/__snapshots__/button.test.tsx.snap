// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match the snapshot 1`] = `
.emotion-0 {
  cursor: pointer;
  -webkit-transition: filter 0.3s ease-in-out;
  transition: filter 0.3s ease-in-out;
}

.emotion-0:hover {
  -webkit-filter: brightness(0.9);
  filter: brightness(0.9);
}

<button
  className="emotion-0"
  onClick={[MockFunction]}
>
  Button
</button>
`;
